flink-runtime/src/test/java/org/apache/flink/runtime/state/TestLocalDirectoryProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.clusterframework.types.AllocationID;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+
+import java.io.File;
+
+/**
+ * Easy-to-construct implementation of {@link LocalRecoveryDirectoryProvider} to use in tests.
+ */
+public class TestLocalDirectoryProvider extends LocalRecoveryDirectoryProviderImpl {
+
+	public TestLocalDirectoryProvider() {
+		super(
+			new File(System.getProperty("java.io.tmpdir")),
```

Comment:
I wouldn't use the tmp dir, but instead pass in a pass. Then the user can instantiate `TemporaryFolder` to make sure that things are cleaned up.